packages/x-charts/src/ChartsXAxis/ChartsGroupedXAxis.tsx

```diff
@@ -0,0 +1,242 @@
+import { isBandScale } from '../internals/isBandScale';
+import { useChartContext } from '../context/ChartProvider/useChartContext';
+import { useXAxes } from '../hooks/useAxis';
+import { getDefaultBaseline, getDefaultTextAnchor } from '../ChartsText/defaultTextPlacement';
+import { invertTextAnchor } from '../internals/invertTextAnchor';
+import { defaultProps, TICK_LABEL_GAP, XAxisRoot, useUtilityClasses } from './utilities';
+import { useTicksGrouped } from '../hooks/useTicksGrouped';
+
+const DEFAULT_GROUPING_CONFIG = {
+  tickSize: 6,
+};
+
+const calculateTickSize = (
+  groupIndex: number,
+  tickSize: number,
+  isConfigArray: boolean = false,
+): number => {
+  // If the groupingConfig is an array we expect the user to provide a `tickSize` for each group.
+  if (isConfigArray) {
+    return tickSize;
+  }
+
+  // Else if it is an object, the provided `tickSize` applies to all groups incrementally.
+  // The first tick will be at `tickSize`, while every subsequent group will be
+  // multiplied by the group index times two and summed to the first tick size.
+  // This allows for a consistent spacing between groups.
+  return tickSize * groupIndex * 2 + tickSize;
+};
+
+const getGroupingConfig = (
+  groupingConfig: AxisGrouping,
+  groupIndex: number,
+  tickSize: number | undefined,
+) => {
+  const config = groupingConfig?.config?.[groupIndex] ?? {};
+
+  return {
+    ...DEFAULT_GROUPING_CONFIG,
+    ...groupingConfig,
+    ...config,
+    tickSize: calculateTickSize(
+      groupIndex,
+      config?.tickSize ?? tickSize ?? DEFAULT_GROUPING_CONFIG.tickSize,
+      Array.isArray(groupingConfig.config),
+    ),
+  };
+};
+
+/**
+ * @ignore - internal component.
+ */
+function ChartsGroupedXAxis(inProps: ChartsXAxisProps) {
+  const { xAxis, xAxisIds } = useXAxes();
+  const { scale: xScale, tickNumber, reverse, ...settings } = xAxis[inProps.axisId ?? xAxisIds[0]];
+
+  if (!isBandScale(xScale)) {
+    throw new Error(
+      'MUI X Charts: ChartsGroupedXAxis only supports the `band` and `point` scale types.',
+    );
+  }
+
+  // eslint-disable-next-line material-ui/mui-name-matches-component-name
+  const themedProps = useThemeProps({ props: { ...settings, ...inProps }, name: 'MuiChartsXAxis' });
+
+  const defaultizedProps = {
+    ...defaultProps,
+    ...themedProps,
+  };
+
+  const {
+    position,
+    disableLine,
+    disableTicks,
+    tickLabelStyle,
+    label,
+    labelStyle,
+    tickSize,
+    valueFormatter,
+    slots,
+    slotProps,
+    tickInterval,
+    tickPlacement,
+    tickLabelPlacement,
+    sx,
+    offset,
+    height: axisHeight,
+  } = defaultizedProps;
+
+  const groupingConfig = (defaultizedProps as { grouping: AxisGrouping }).grouping;
+
+  const theme = useTheme();
+  const isRtl = useRtl();
+  const classes = useUtilityClasses(defaultizedProps);
+  const drawingArea = useDrawingArea();
+  const { left, top, width, height } = drawingArea;
+  const { instance } = useChartContext();
```

Comment:
A lot of code in this file seems to be duplicated in `ChartsSingleXAxis`. Could we find a way to reuse it? Otherwise we'll need to remember that we need to apply changes in both files simultaneously and I'm a bit afraid we might create some bugs due to forgetting to do that.

Reply:
Why do you say it would only be marginally better?

Reply:
I initially though a lot of logic had to live in the ocmponent anyways, but I could reorganize the hooks to streamline them